### ISSUE TITLE
`DomainItemAccessibleObject` and `DomainUpDownAccessibleObject` should be marked as "obsolete"

### ISSUE THREAD

**Issue by vladimir-krestov** [dotnet/winforms#7344]
### Is your feature request related to a problem? Please describe

Related to fix #7330. Based on the [comment](https://github.com/dotnet/winforms/pull/7330#pullrequestreview-1017923802).
- `DomainItemAccessibleObject` class is useless, we don't need it to build `DomainUpDown` accessibility tree, because we don't show "Items" list anymore (see #4207).
- `DomainUpDownAccessibleObject` class is useless, we don't need it, because the base `UpDownBaseAccessibleObject` (from #7330) has all necessary implementation.

We might to remove them, but we can't because they are "public". 

### Describe the solution you'd like and alternatives you've considered

From the [comment](https://github.com/dotnet/winforms/pull/7330#pullrequestreview-1017923802):
> ...we should obsolete the two defunct types with `[ObsoleteAttribute(..., IsError = false)]`...

### Will this feature affect UI controls?

No, we don't use these classes for building of `DomainUpDown` accessibility tree.

**Comment by vladimir-krestov** [comment]
@RussKie, FYI

**Comment by vladimir-krestov** [comment]
Closed as completed in #7330
